lib/index.d.ts

```diff
@@ -2,474 +2,596 @@
-    getAutoClearForces(): boolean;
-    clearForces(): void;
-    queryAABB(aabb: AABB, queryCallback: (fixture: Fixture) => boolean): void;
-    rayCast(point1: Vec2, point2: Vec2, reportFixtureCallback: (fixture: Fixture, point: Vec2, normal: Vec2, fraction: number) => number): void;
-    getProxyCount(): number;
-    getTreeHeight(): number;
-    getTreeBalance(): number;
-    getTreeQuality(): number;
-    shiftOrigin(newOrigin: Vec2): void;
-    createBody(def: BodyDef): Body;
-    createBody(position: Vec2, angle?: number): Body;
-    createBody(): Body;
-    createDynamicBody(def: BodyDef): Body;
-    createDynamicBody(position: Vec2, angle?: number): Body;
-    createDynamicBody(): Body;
-    createKinematicBody(def: BodyDef): Body;
-    createKinematicBody(position: Vec2, angle?: number): Body;
-    createKinematicBody(): Body;
-    destroyBody(b: Body): boolean;
-    createJoint<T extends Joint>(joint: T): T | null;
-    destroyJoint(joint: Joint): void;
-    step(timeStep: number, velocityIterations?: number, positionIterations?: number): void;
-    findNewContacts(): void;
-    /**
-     * @private
-     */
-    createContact(proxyA: FixtureProxy, proxyB: FixtureProxy): void;
-    updateContacts(): void;
-    destroyContact(contact: Contact): void;
-
-    _listeners: any; // TODO
-
-    on(name: 'begin-contact', listener: (contact: Contact) => void): World;
-    on(name: 'end-contact', listener: (contact: Contact) => void): World;
-    on(name: 'pre-solve', listener: (contact: Contact, oldManifold: Manifold) => void): World;
-    on(name: 'post-solve', listener: (contact: Contact, impulse: ContactImpulse) => void): World;
-    on(name: 'remove-body', listener: (body: Body) => void): World;
-    on(name: 'remove-joint', listener: (joint: Joint) => void): World;
-    on(name: 'remove-fixture', listener: (fixture: Fixture) => void): World;
-    off(name: 'begin-contact', listener: (contact: Contact) => void): World;
-    off(name: 'end-contact', listener: (contact: Contact) => void): World;
-    off(name: 'pre-solve', listener: (contact: Contact, oldManifold: Manifold) => void): World;
-    off(name: 'post-solve', listener: (contact: Contact, impulse: ContactImpulse) => void): World;
-    off(name: 'remove-body', listener: (body: Body) => void): World;
-    off(name: 'remove-joint', listener: (joint: Joint) => void): World;
-    off(name: 'remove-fixture', listener: (fixture: Fixture) => void): World;
-
-    publish(name: string, arg1: any, arg2: any, arg3: any): number;
-
-    beginContact(contact: Contact): void;
-    endContact(contact: Contact): void;
-    preSolve(contact: Contact, oldManifold: Manifold): void;
-    postSolve(contact: Contact, impulse: ContactImpulse): void;
-  }
-
-  let World: {
-    new(def: WorldDef): World;
-    (def: WorldDef): World;
-
-    new(gravity: Vec2): World;
-    (gravity: Vec2): World;
-
-    new(): World;
-    (): World;
-  };
+export as namespace planck;
```

Comment:
what would happen if we don't export a namespace?

Reply:
You use the namespace when you are in a non-module environment. That means you just write `import "planck-js";` on top of a file or add `"planck-js"` to the `types` field in the tsconfig file to make TS aware of the typings. In this case TS assumes there is a global `planck` object.
So if you don't export the namespace you can only use the typings in a module environment.